query-engine/connectors/sql-query-connector/src/cursor_condition.rs

```diff
@@ -229,20 +311,105 @@ fn map_equality_condition(field: &ScalarFieldRef) -> Expression<'static> {
     }
 }
 
-fn order_definitions(query_arguments: &QueryArguments, model: &ModelRef) -> Vec<(ScalarFieldRef, SortOrder)> {
-    let defined_ordering: Vec<_> = query_arguments
-        .order_by
-        .iter()
-        .map(|o| (o.field.clone(), o.sort_order))
-        .collect();
+fn order_definitions(
+    query_arguments: &QueryArguments,
+    model: &ModelRef,
+    ordering_joins: &Vec<OrderingJoins>,
+) -> Vec<CursorOrderDefinition> {
+    let mut orderings: Vec<CursorOrderDefinition> = vec![];
+
+    for (index, order_by) in query_arguments.order_by.iter().enumerate() {
+        let (last_hop, before_last_hop) = take_last_two_elem(&order_by.path);
+        let joins_for_hop = ordering_joins.get(index).unwrap();
```

Comment:
That's how we read the joins in the cursor_condition method. I feel a bit of danger here, as if there's ever a change in `ordering::build` in the way joins are computed and there's no longer as many `ordering_joins` as there are `query_arguments.order_by`, this will break.

We somehow have loose assumptions between the two methods and I can see many ways it could go wrong in the future. Tests are here to catch issues anyway, but just so you know.

Reply:
You are right, this is not great. I feel like an overhaul of cursor / order building is necessary to bring it closer together. For now, tests catching issues is sufficient.